.circleci/config.yml

```diff
@@ -304,6 +318,38 @@ jobs:
             - mattermost-server
             - mattermost-webapp
 
+  build-focalboard:
+    docker:
+      - image: mattermost/mattermost-build-server:20220415_golang-1.18.1
+    resource_class: xlarge
+    working_directory: ~/mattermost
+    steps:
+      - attach_workspace:
+          at: ~/mattermost/
+      - run:
+          command: |
+            cd focalboard
+            make server-linux
+  configure-multirepo:
+    docker:
+      - image: mattermost/mattermost-build-server:20220415_golang-1.18.1
+    working_directory: ~/mattermost
+    steps:
+      - attach_workspace:
+          at: ~/mattermost/
+      - run:
+          command: |
+            cd mattermost-server
+            make setup-go-work
+            cd ../focalboard
+            EXCLUDE_ENTERPRISE=TRUE make templates-archive
```

Comment:
The `EXCLUDE_ENTERPRISE` env var is only used by the `setup-go-work` task.  It should be added there instead?

Reply:
@phoinixgrr , this doesn't make sense to me.  While I do understand how makefile prerequisites work, I don't see how adding `EXCLUDE_ENTERPRISE` to the `make templates-archive` has any effect.

In the `make setup-go-work` task two lines above, a go.work file will be created.  If `EXCLUDE_ENTERPRISE` is already defined somewhere as true then it will not include enterprise repo, otherwise it will.  Regardless, the file will be created (if not already existing).  Once it gets to the `make templates-archive` task, the go.work file already exists and won't be recreated because the `-f` flag (force) is not set, therefore the `EXCLUDE_ENTERPRISE` is ignored.

If you want the enterprise repo excluded then that var should appear prefixed to `make setup-go-work`, otherwise it should be removed altogether.

Reply:
@wiggin77  

Thanks for the feedback. 

Part of the PR deliverables is to successfully parallel build focalboard and mattermost-server. 
![Screenshot 2022-08-09 at 16 56 10](https://user-images.githubusercontent.com/36594156/183667710-907188e5-6cd7-482b-ba91-777082beddde.png)
 
During our testing, without running the `templates-archive` target (which also runs the `setup-go-work`) the build fails.

Reply:
Would you be ok with changing that to: 
```
 EXCLUDE_ENTERPRISE=TRUE make setup-go-work 
 make templates-archive
```

Or you need the `make templates-archive` step moved to a different stage? 
Let me know.

Reply:
Thanks for the comment @wiggin77  

I see that the Makefile is [defined by default](https://github.com/mattermost/focalboard/blob/main/Makefile#L42)  in the Makefile . Therefore setting `EXCLUDE_ENTERPRISE=true` in my proposal  can actually be omitted
(Just hope that the upstream makefile does not change this behaviour in the future, cause if it does it will brake this build. Therefore it is a good practise defining this). 

Concerning the argument of exporting the Environment variable `EXCLUDE_ENTERPRISE` on the  `templates-archive` target, instead of the `setup-go-work` , being ignored, that's not actually the case. 

Example checkingout Focalboard repository
```
❯ rm go.work
❯ EXCLUDE_ENTERPRISE=FALSE  make templates-archive
go run ./build/gowork/main.go
go.work written successfully.
go 1.18

use ./server
use ../mattermost-server
use ../enterprise
use ./mattermost-plugin

cd server/assets/build-template-archive; go run -tags 'json1' main.go --dir="../templates-boardarchive" --out="../templates.boardarchive"
```

It is clearly shown, that running the `make templates-archive` target with env variable set to: `EXCLUDE_ENTERPRISE=FALSE` , `setup-go-work` that is being called automatically as dependancy, it clearly automatically sets correctly the variable (EXCLUDE_ENTERPRISE is not ignored) , as it has printed `use ../enterprise` 

---
To sum this up. 

Would you be ok with changing this to 
```
 make templates-archive
 ``` 

Just to remind, the goal is to: 
1. Setup the go.work file without enterprise repo
2. run  `templates-archive` process

Reply:
@wiggin77  Any news on this?